pkg/upstream/cluster/cluster_manager.go

```diff
@@ -570,15 +589,29 @@ func (cm *clusterManager) getActiveConnectionPool(balancerContext types.LoadBala
 
 func (cm *clusterManager) ShutdownConnectionPool(proto types.ProtocolName, addr string) {
 	shutdown := func(value interface{}) {
-		connectionPool := value.(*sync.Map)
-		if connPool, ok := connectionPool.Load(addr); ok {
-			pool := connPool.(types.ConnectionPool)
-			connectionPool.Delete(addr)
-			pool.Shutdown()
-			if log.DefaultLogger.GetLogLevel() >= log.DEBUG {
-				log.DefaultLogger.Debugf("[upstream] [cluster manager] protocol %s address %s connections shutdown", proto, addr)
+		// value is {addr: pool} or {clusterName: {addr: pool}}
+		cm.clustersMap.Range(func(clusterName, _ interface{}) bool {
```

Comment:
`Range` operation is very inefficient with a large number of clusters, and maybe we could add a type judgement to this map.